src/Bicep.Decompiler/TemplateConverter.cs

```diff
@@ -613,10 +613,16 @@ private SyntaxBase ParseString(string? value, IJsonLineInfo lineInfo)
             }
         }
 
-        private static IntegerLiteralSyntax ParseIntegerJToken(JValue value)
+        private static ExpressionSyntax ParseIntegerJToken(JValue value)
         {
-            var ulongValue = value.Value<ulong>();
-            return SyntaxFactory.CreateIntegerLiteral(ulongValue);
+            var longValue = value.Value<long>();
+            if (longValue < 0)
+            {
+                return SyntaxFactory.CreateNegativeIntegerLiteral((ulong)longValue);
```

Comment:
Please add an integration test for this - see https://github.com/Azure/bicep/blob/main/src/Bicep.Decompiler.IntegrationTests/Working/issue5881/main.json for an example.

Reply:
Added!